src/ol_infrastructure/applications/ocw_build/__main__.py

```diff
@@ -0,0 +1,220 @@
+)
+
+# Instance profile that includes access to that bucket, plus others
+ocw_next_instance_policy_data = {
+    'Version': '2012-10-17',
+    'Statement': [
+        {
+            'Effect': 'Allow',
+            'Action': [
+                's3:List*',
+                's3:Get*'
+            ],
+            'Resource': [
+                # This bucket is not managed here, but we need access:
+                'arn:aws:s3:::open-learning-course-data-*'
+            ]
+        },
+        {
+            'Effect': 'Allow',
+            'Action': [
+                's3:GetAccessPoint',
+                's3:GetAccountPublicAccessBlock',
+                's3:ListAllMyBuckets',
+                's3:ListAccessPoints',
+                's3:ListJobs',
+                's3:HeadBucket'
+            ],
+            'Resource': '*'
+        },
+        {
+            'Effect': 'Allow',
+            'Action': [
+                's3:List*',
+                's3:Get*',
+                's3:Put*',
+                's3:Delete*'
+            ],
+            'Resource': [
+                f'arn:aws:s3::{website_bucket_name}',
+                f'arn:aws:s3::{website_bucket_name}/*'
+            ]
+        }
+    ]
+}
+iam_instance_policy = iam.Policy(
+    f'ocw-build-instance-policy-{env_suffix}',
+    name=f'ocw-build-instance-policy-{env_suffix}',
+    path=f'/ol-applications/ocw-build-policy-{env_suffix}',
+    policy=ocw_next_instance_policy_data,
+    description='Grants access to S3 buckets from the OCW Build server'
+)
+
+iam_role = iam.Role(
+    f'ocw-build-instance-role-{env_suffix}',
+    assume_role_policy=(
+        {
+            'Version': '2012-10-17',
+            'Statement': {
+                'Effect': 'Allow',
+                'Action': 'sts:AssumeRole',
+                'Principal': {'Service': 'ec2.amazonaws.com'}
+            }
+        }
+    ),
+    name=f'ocw-build-instance-role-{env_suffix}',
+    path='/ol-applications/ocw-build-role'
+)
+
+iam.RolePolicyAttachment(
+    f'ocw-build-role-policy-{env_suffix}',
+    policy_arn=iam_instance_policy.arn,
+    role=iam_role.name
+)
+
+ocw_build_instance_profile = iam.InstanceProfile(
+    f'ocw-build-instance-profile-{env_suffix}',
+    role=iam_role.name,
+    name=f'ocw-build-instance-profile-{env_suffix}',
+    path='/ol-applications/ocw-build-instance-profile/'
+)
+
+# Salt minion
+
+ocw_build_minion_id = f'ocw-build-{ocw_next_build_environment}-0'
+salt_minion = OLSaltStackMinion(
+    f'saltstack-minion-{ocw_build_minion_id}',
+    OLSaltStackMinionInputs(
+        minion_id=ocw_build_minion_id,
+        salt_api_url=get_config('saltstack:api_url'),
+        salt_user=get_config('saltstack:api_user'),
+        salt_password=get_config('saltstack:api_password')
+    )
+)
+
+cloud_init_userdata = build_userdata(
+    instance_name=ocw_build_minion_id,
+    minion_keys=salt_minion,
+    minion_roles=['ocw-build'],
+    minion_environment=ocw_next_build_environment,
```

Comment:
This will need to match the `environment` grain that we are using in the salt code. For the apps VPCs we'll likely want to do some migration to align naming between projects. For this particular server, assuming that there are no pillar or state values that rely on the environment grain then we should probably keep this value as-is.

Reply:
This instance is getting deployed in the apps VPCs, so the name used in Salt currently is `rc-apps` and `production-apps`, vs `applications-qa` and `applications-production` in Pulumi. As long as the salt code for the OCW build machine isn't relying on the environment grain anywhere then I think we should leave it as is and we should plan to migrate the naming in the Salt code to use the format that we defined here.